bitwarden_license/bitwarden-sm/src/secrets/create.rs

```diff
@@ -34,16 +34,23 @@ pub(crate) async fn create_secret(
 ) -> Result<SecretResponse, SecretsManagerError> {
     input.validate()?;
 
-    let enc = client.internal.get_encryption_settings()?;
-    let key = enc.get_key(&Some(input.organization_id))?;
+    let key_store = client.internal.get_key_store();
+    let key = SymmetricKeyId::Organization(input.organization_id);
+    let mut ctx = key_store.context();
 
     let secret = Some(SecretCreateRequestModel {
-        key: input.key.clone().trim().encrypt_with_key(key)?.to_string(),
-        value: input.value.clone().encrypt_with_key(key)?.to_string(),
-        note: input.note.clone().trim().encrypt_with_key(key)?.to_string(),
+        key: input.key.clone().trim().encrypt(&mut ctx, key)?.to_string(),
+        value: input.value.clone().encrypt(&mut ctx, key)?.to_string(),
+        note: input
+            .note
+            .clone()
+            .trim()
+            .encrypt(&mut ctx, key)?
+            .to_string(),
         project_ids: input.project_ids.clone(),
         access_policies_requests: None,
     });
+    drop(ctx);
```

Comment:
Is this to avoid keeping them in memory longer than needed? Can we document it if that's the case.

Would it be nicer to use a block scope?

```
let secret = {
    let mut ctx = key_store.context();

    Some(SecretCreateRequestModel {
        key: input.key.clone().trim().encrypt(&mut ctx, key)?.to_string(),
        value: input.value.clone().encrypt(&mut ctx, key)?.to_string(),
        note: input
            .note
            .clone()
            .trim()
            .encrypt(&mut ctx, key)?
            .to_string(),
        project_ids: input.project_ids.clone(),
        access_policies_requests: None,
    })
}; 
```

Reply:
Because we're using RwLocks inside the context, if you hold on to them across an await point, that makes the async future !Send, which causes compilation issues in the wasm and uniffi bindings.

Dropping the context before we call the API endpoint is one way to solve it, I agree that looks nicer. I'll change it.

I'll add some docs mentioning that it's not recommended to hold the context across await points too.

Reply:
Resolved in https://github.com/bitwarden/sdk-internal/commit/d454cf4c0228d0c0921662c603347aa082de324f?w=1